### PR TITLE
Add binary sensor icon for DEVICE_CLASS_BATTERY_CHARGING

### PR DESCRIPTION
**Pull request by epenet**
## Proposed change

[DEVICE_CLASS_BATTERY_CHARGING ](https://github.com/home-assistant/core/blob/dev/homeassistant/components/binary_sensor/__init__.py#L29) does not currently have associated icons.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/common/entity/binary_sensor_icon.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ export const binarySensorIcon = (state: HassEntity) => {
   switch (state.attributes.device_class) {
     case "battery":
       return activated ? "hass:battery" : "hass:battery-outline";
+    case "battery_charging":
+      return activated ? "hass:flash" : "hass:flash-off";
```
Comment (ludeeus) [inline src/common/entity/binary_sensor_icon.ts:11]: `battery-charging` / `battery` ?

Comment (epenet) [reply src/common/entity/binary_sensor_icon.ts:11]: According to the [binary_sensor](https://github.com/home-assistant/core/blob/dev/homeassistant/components/binary_sensor/__init__.py):
```python
# On means low, Off means normal
DEVICE_CLASS_BATTERY = "battery"

# On means charging, Off means not charging
DEVICE_CLASS_BATTERY_CHARGING = "battery_charging"
```

Comment (ludeeus) [reply src/common/entity/binary_sensor_icon.ts:11]: Right, what I'm saying is use the battery-charging icon.
flash is overused, and here there are dedicated icons for the purpose

Comment (epenet) [reply src/common/entity/binary_sensor_icon.ts]: Icon set updated

File: src/common/entity/binary_sensor_icon.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ export const binarySensorIcon = (state: HassEntity) => {
   switch (state.attributes.device_class) {
     case "battery":
       return activated ? "hass:battery" : "hass:battery-outline";
+    case "battery_charging":
+      return activated ? "hass:battery" : "hass:battery-charging";
```
Comment (ludeeus) [inline src/common/entity/binary_sensor_icon.ts:11]: These should be switched right?
Now it shows `hass:battery-charging` when not charging

Comment (epenet) [reply src/common/entity/binary_sensor_icon.ts:11]: No, I think it's correct as it is.
```python
  const activated = state.state && state.state === "off";
```

So if state is off, show battery else show battery-charging

Comment (ludeeus) [reply src/common/entity/binary_sensor_icon.ts:11]: wtf... yeah :+1:

**Review by ludeeus — APPROVED** [review]

**Pull request merged**